File: stocks_monitor_agent/src/model/stock_data.rs
use serde::{Deserialize, Serialize};

/// A stock or index being monitored
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MonitoredStock {
    pub ticker: String,
}

/// A closing price record for a given ticker on a given date
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct StockClosingPrice {
    pub ticker: String,
    pub date: String,
    pub closing_price: f64,
}

/// The result of evaluating alert rules for a single ticker
#[derive(Debug, Clone)]
pub struct StockAlertEvaluation {
    pub ticker: String,
    pub daily_change_percent: Option<f64>,
    pub weekly_change_percent: Option<f64>,
    pub daily_alert_triggered: bool,
    pub weekly_alert_triggered: bool,
}

File: stocks_monitor_agent/app/model/stock_data.py
from dataclasses import dataclass


@dataclass
class MonitoredStock:
    """A stock or index being monitored."""
    ticker: str


@dataclass
class StockMarketData:
    """Price and change data for a ticker fetched via AgentCore Browser."""
    ticker: str
    current_price: float
    daily_change_percent: float
    five_day_change_percent: float
    thirty_day_change_percent: float
    three_month_change_percent: float
    ytd_change_percent: float
    one_year_change_percent: float
